pkg/kubelet/metrics/metrics.go

```diff
@@ -211,6 +213,30 @@ var (
 		[]string{},
 	)
 
+	// FirstNetworkPodStartSLIDuration is a gauge that tracks the duration (in seconds) it takes for the first network pod to run,
+	// excluding the time for image pulling. This metric should reflect the "Pod startup latency SLI" definition and measure the time
+	// for the CNI plugin to be able to provide the network capabilities to the Pod.
+	// ref: https://github.com/kubernetes/community/blob/master/sig-scalability/slos/pod_startup_latency.md
+	FirstNetworkPodStartSLIDuration = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           FirstNetworkPodStartSLIDurationKey,
+			Help:           "Duration in seconds to start the first network pod, excluding time to pull images and run init containers, measured from pod creation timestamp to when all its containers are reported as started and observed via watch",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// FirstNetworkPodStartTotalDuration is a gauge that tracks the duration (in seconds) it takes for the first network pod to run
+	// since creation, including the time for image pulling.
```

Comment:
I don't think it is necessary to have both metrics with and without accounting for the image pulling and init containers latency. As far as I understand the context behind this PR, having a metric dedicated strictly to network initialization latency is what we want in order to detect late startup caused by CNI initialization.

Reply:
I think we should only keep the metric that doesn't account for image pulling and init container latency which is the SLI one, because we already have `pod_start_total_duration_seconds` to measure the latency seen by the users. Here what we specifically want to know is whether this latency is caused by the network initialization on the node or not. So the less extraneous latency it accounts for, the better.

Reply:
done